Public/Src/Engine/Cache/Artifacts/LocalDiskContentStore.cs

```diff
@@ -313,11 +313,7 @@ private Possible<Unit> CreateSymlinkIfNotExistsOrTargetMismatch(AbsolutePath sym
                 var updatedDestination = destination.Substring(0, destination.Length - requiredFileName.Length) + requiredFileName;
 
                 // Move file to temporary location
-                var changeCasing = await FileUtilities.MoveFileAsync(destination, updatedDestination, replaceExisting: false);
-                if (!changeCasing)
-                {
-                    return new Failure<string>(I($"Could not move file '{destination}' to change file name casing to '{requiredFileName}'."));

```

Comment:
>    return new Failure<string>(I($"Could not move file '{destination}' to change file name casing to '{requiredFileName}'.")); [](start = 16, length = 126)

You changed the behavior here from returning a Failure<> to throwing an exception. Is that safe to do? I think it's unexpected for a Possible<>returning method to throw sometimes and return a Failure other times

Reply:
Previously, the failure code paths in all occurrences of `MoveFileAsycn` have never been exercised. `MoveFileAsync` either returns true (not `Possible` nor `Failure`) or throws an exception. Also, this consistent with `MoveFile` method in .NET. 

I couldn't change `CopyFileAsync` because our version take a predicate as additional argument.

Reply:
I was referring to `Task<Possible<Unit, Failure>> EnsureFileNameCasingMatchAsync()` now throwing rather than returning a Failure. Now that I look closer I guess it won't actually throw because `MoveFileAsync()` wraps its `NativeWin32Exception` in a `BuildXLException` which `EnsureFileNameCasingMatchAsync` then handles. So nevermind on that one